### PR TITLE
Add Log4J 2 Logging Support

### PR DESCRIPTION
**Pull request by ThatJoeMoore**
Note that I left out the call to LambdaRuntimeInternal.setUseLog4jAppender, as I don't know what effect that has and what consequences including it would have.

### CODE REVIEW

**Comment by zodeus** [comment]
Have you tested this? I copied this class into my project, log4j seems to find it fine and route through it if I invoke the jar locally. However when I upload the fat jar up to aws I throws this error 
```
ERROR StatusLogger Unrecognized format specifier [d]                                                       
ERROR StatusLogger Unrecognized conversion specifier [d] starting at position 16 in conversion pattern.    
ERROR StatusLogger Unrecognized format specifier [thread]                                                  
ERROR StatusLogger Unrecognized conversion specifier [thread] starting at position 25 in conversion pattern
ERROR StatusLogger Unrecognized format specifier [level]                                                   
ERROR StatusLogger Unrecognized conversion specifier [level] starting at position 35 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [logger]                                                  
ERROR StatusLogger Unrecognized conversion specifier [logger] starting at position 47 in conversion pattern
ERROR StatusLogger Unrecognized format specifier [msg]                                                     
ERROR StatusLogger Unrecognized conversion specifier [msg] starting at position 54 in conversion pattern.  
ERROR StatusLogger Unrecognized format specifier [n]                                                       
ERROR StatusLogger Unrecognized conversion specifier [n] starting at position 56 in conversion pattern.    
ERROR StatusLogger No log4j2 configuration file found. Using default configuration: logging only errors to 
ERROR StatusLogger Unrecognized format specifier [d]                                                       
ERROR StatusLogger Unrecognized conversion specifier [d] starting at position 16 in conversion pattern.    
ERROR StatusLogger Unrecognized format specifier [thread]                                                  
ERROR StatusLogger Unrecognized conversion specifier [thread] starting at position 25 in conversion pattern
ERROR StatusLogger Unrecognized format specifier [level]                                                   
ERROR StatusLogger Unrecognized conversion specifier [level] starting at position 35 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [logger]                                                  
ERROR StatusLogger Unrecognized conversion specifier [logger] starting at position 47 in conversion pattern
ERROR StatusLogger Unrecognized format specifier [msg]                                                     
ERROR StatusLogger Unrecognized conversion specifier [msg] starting at position 54 in conversion pattern.  
ERROR StatusLogger Unrecognized format specifier [n]                                                       
ERROR StatusLogger Unrecognized conversion specifier [n] starting at position 56 in conversion pattern.    
END RequestId: 519d69b1-7723-11e6-80a1-e765f3e42e38                                                        
```

**Comment by williamblevins** [comment]
Great timing with this patch. I was looking to do this just this morning. I tried this and it worked as expected! I needed to make one small tweak with the default encoding. Much appreciated.

File: aws-lambda-java-log4j2/src/main/java/com/amazonaws/services/lambda/runtime/log4j/LambdaAppender.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+public class LambdaAppender extends AbstractAppender {
+
+    LambdaLogger logger = LambdaRuntime.getLogger();
+
+    private LambdaAppender(String name, Filter filter, Layout<? extends Serializable> layout, boolean ignoreExceptions) {
+        super(name, filter, layout, ignoreExceptions);
+    }
+
+    public void append(LogEvent event) {
+        logger.log(new String(getLayout().toByteArray(event)));
```
Comment (williamblevins) [inline aws-lambda-java-log4j2/src/main/java/com/amazonaws/services/lambda/runtime/log4j/LambdaAppender.java]: Recommend ensuring some sort of reasonable encoding here: UTF-8?

**Comment by vg-appsdirect** [comment]
Guys, let's merge this PR. We need log4j2!

**Comment by asuresh8** [comment]
Added support in last commit

**Pull request closed without merging**